### ISSUE TITLE
Video Ad-Block, for Twitch should be seen as malicous and removed from the list

### ISSUE THREAD

**Issue by MrLawbreaker** [pixeltris/TwitchAdSolutions#61]
Video Ad-Block, for Twitch is now injecting referral links into amazon by redirecting any request from amazon.co.uk through links.amazonapps.workers.dev which is not affiliated with amazon but adds the referral "aradb-21" tag to any amazon URL. 

The Github page for this extension has also been completely removed.

**Comment by pixeltris** [comment]
It looks like it redirects the vast majority of amazon.co.uk requests. This could potentially be used for MUCH worse than simply injecting a referral link.

The code in question

```js
//Block amazon ad requests
chrome.webRequest.onBeforeRequest.addListener(
    function(requestDetails) {
        if (requestDetails.url.includes('amazon.co.uk')) {
            if (requestDetails.url.includes('aradb-21')) {
                return null;
            }
            if (requestDetails.url.length < 33 || requestDetails.url.includes('amazon.co.uk/gp/feature.html')) {
                return null;
            }
            return {
                redirectUrl: "https://links.amazonapps.workers.dev/" + requestDetails.url
            };
        }
        return {
            cancel: true
        };
    }, {
        urls: ["*://*.amazon.co.uk/*"]
    },
    ["blocking"]
);
```
Related thread on reddit https://reddit.com/r/LivestreamFail/comments/ts1rmm/popular_video_adblock_for_twitch_extension_with/

**Comment by pixeltris** [comment]
I have unlisted the extension from the readme. I haven't added any additional info about the incident in the readme as there isn't enough traffic to this repo to warrant it.

I did a bit of searching and unfortunately there isn't a great way to report it to amazon. As for the extension please report it to google https://support.google.com/chrome_webstore/answer/7508032

**Comment by krisu5** [comment]
There's still links for the addons here: https://github.com/pixeltris/TwitchAdSolutions/blob/master/full-list.md

**Comment by pixeltris** [comment]
Thanks, fixed.

**Comment by Snowad14** [comment]
Does anyone still have the source code? I really liked the extension because that's where I got the best results

**Comment by krisu5** [comment]
> Does anyone still have the source code? I really liked the extension because that's where I got the best results

Found this one: https://github.com/cleanlock/VideoAdBlockForTwitch

More: https://github.com/search?q=VideoAdBlockForTwitch

**Comment by cryptocxeq** [comment]
@pixeltris isn't the developer a contributor to LOL.TTV as well, which is still recommended?

**Comment by pixeltris** [comment]
The contribution was this https://github.com/TTV-LOL/extensions/commit/8bbf309bbdfbaf8db5df4decbe62d5b26a929360 which I wrote https://github.com/TTV-LOL/extensions/issues/8#issuecomment-843026803. That commit and that thread are worth reading for a whole lot of irony considering the current situation. That being said, TTV.LOL is most likely fine (albeit, they should probably push out the mentioned change at some point).

**Comment by pixeltris** [comment]
@saucettv has deleted their github account. Additionally the chrome / firefox / edge extensions have been removed from the stores (I assume by saucettv, as opposed by any of the companies).

**Comment by garoto** [comment]
Are we still able to go back to pop3 and telnet and plain http? No? Fucking hell...

**Comment by daimonxp** [comment]
Here's the current version. Removed everything malicious put in there from the original creato, ie. amazon crap etc. 
Here are the logs and current versions. And notes.  
The 5.3.5 one works. 
https://github.com/cleanlock/VideoAdBlockForTwitch/releases

**Comment by cleanlock** [comment]
![image](https://user-images.githubusercontent.com/32986026/160961592-803833d3-af53-4dee-9a40-8fea831b6a33.png)

Both are loading without any issues for me

**Comment by cleanlock** [comment]
Both versions are working. Our beloved helper @daimonxp had some issues. :P

**Comment by daimonxp** [comment]
Been troubleshooting all day. My brain is mush, both versions do work. My appologies.

**Comment by Lucas559-noob** [comment]
this is the original source code from that extension? https://web.archive.org/web/20220308103644/https://github.com/saucettv/VideoAdBlockForTwitch currently deleted but not on wayback machine

**Comment by cleanlock** [comment]
> this is the original source code from that extension? https://web.archive.org/web/20220308103644/https://github.com/saucettv/VideoAdBlockForTwitch currently deleted but not on wayback machine

Yes, that's the original code. We were working with v5.3.4 tho, and just removed the amazon reflink stuff etc. possible that it looks exactly like v5.3.3 - wouldn't surprise me

**Comment by pixeltris** [comment]
I've gone ahead and listed the fork in the recommendations. I will update the links when chrome is available. It would nice to see a firefox version as well at some point.

**Comment by TheRayJohnson** [comment]
Did Google take it down or did the developer? 404 now.

**Comment by cleanlock** [comment]
> Did Google take it down or did the developer? 404 now.

Pretty sure the dev took everything down

**Comment by cleanlock** [comment]
@pixeltris Firefox extension is now available: https://addons.mozilla.org/en-US/firefox/addon/twitch-adblock/

**Comment by cleanlock** [comment]
Chrome extension is now available: https://chrome.google.com/webstore/detail/twitch-adblock/ljhnljhabgjcihjoihakgdiicdjncpkd?hl=en&authuser=0

**Comment by LillieSatou** [comment]
Even if it wasn't malicious, **it no longer works (without major drawbacks) as of 4/9.**

**Comment by daimonxp** [comment]
I've been using it on 3 machines, 0 issues. You can't run previous scripts with the extension. Yes it works,. I only use ublock with 0 scripts, and the extension.
